x-pack/test/api_integration/apis/ml/system/capabilities.ts

```diff
@@ -0,0 +1,120 @@
+      .set(COMMON_REQUEST_HEADERS)
+      .expect(200);
+
+    return body;
+  }
+
+  describe('ml_capabilities', () => {
+    describe('get capabilities', function () {
+      it('should be enabled in space', async () => {
+        const { mlFeatureEnabledInSpace } = await runRequest(USER.ML_POWERUSER);
+        expect(mlFeatureEnabledInSpace).to.eql(true);
+      });
+
+      it('should have upgradeInProgress false', async () => {
+        const { upgradeInProgress } = await runRequest(USER.ML_POWERUSER);
+        expect(upgradeInProgress).to.eql(false);
+      });
+
+      it('should have full license', async () => {
+        const { isPlatinumOrTrialLicense } = await runRequest(USER.ML_POWERUSER);
+        expect(isPlatinumOrTrialLicense).to.eql(true);
+      });
+
+      it('should have the right number of capabilities', async () => {
+        const { capabilities } = await runRequest(USER.ML_POWERUSER);
+        expect(Object.keys(capabilities).length).to.eql(29);
+      });
+
+      it('should get viewer capabilities', async () => {
+        const { capabilities } = await runRequest(USER.ML_VIEWER);
+
+        expect(capabilities.canCreateJob).to.eql(false);
+        expect(capabilities.canDeleteJob).to.eql(false);
+        expect(capabilities.canOpenJob).to.eql(false);
+        expect(capabilities.canCloseJob).to.eql(false);
+        expect(capabilities.canUpdateJob).to.eql(false);
+        expect(capabilities.canForecastJob).to.eql(false);
+        expect(capabilities.canCreateDatafeed).to.eql(false);
+        expect(capabilities.canDeleteDatafeed).to.eql(false);
+        expect(capabilities.canStartStopDatafeed).to.eql(false);
+        expect(capabilities.canUpdateDatafeed).to.eql(false);
+        expect(capabilities.canPreviewDatafeed).to.eql(false);
+        expect(capabilities.canGetFilters).to.eql(false);
+        expect(capabilities.canCreateCalendar).to.eql(false);
+        expect(capabilities.canDeleteCalendar).to.eql(false);
+        expect(capabilities.canCreateFilter).to.eql(false);
+        expect(capabilities.canDeleteFilter).to.eql(false);
+        expect(capabilities.canCreateDataFrameAnalytics).to.eql(false);
+        expect(capabilities.canDeleteDataFrameAnalytics).to.eql(false);
+        expect(capabilities.canStartStopDataFrameAnalytics).to.eql(false);
+        expect(capabilities.canCreateMlAlerts).to.eql(false);
+        expect(capabilities.canAccessML).to.eql(true);
+        expect(capabilities.canGetJobs).to.eql(true);
+        expect(capabilities.canGetDatafeeds).to.eql(true);
+        expect(capabilities.canGetCalendars).to.eql(true);
+        expect(capabilities.canFindFileStructure).to.eql(true);
+        expect(capabilities.canGetDataFrameAnalytics).to.eql(true);
+        expect(capabilities.canGetAnnotations).to.eql(true);
+        expect(capabilities.canCreateAnnotation).to.eql(true);
+        expect(capabilities.canDeleteAnnotation).to.eql(true);
```

Comment:
If one of those capabilities checks fails, it'll be hard to find out which one from the log, as it would be something like `expected false to kind of equal true`. I'd suggest to either add assertion messages like
```
expect(capabilities.canDeleteAnnotation).to.eql(true, 'canDeleteAnnotation should be true');
```
or assert the whole object (which will nicely print out which part didn't match on a failed assertion):
```
expect(capabilities).to.eql({
  canCreateJob: true,
  canDeleteJob: false,
  ...
]);
```
This should work for different order of properties.
The same goes for the other long blocks of property assertions, I won't repeat this comment.

Reply:
i'll add assertion messages. i purposefully didn't compare the entire object as we don't care about the order of the properties.

Reply:
i'll change to assert the whole object